### PR TITLE
Add linear orders

### PR DESCRIPTION
**Pull request by peepopoggers**
image
:cat: :+1: 

This may need some checks but it works for orders (BTCUSDT) :L

### CODE REVIEW

File: src/order.rs [hunk 1]
```diff
@@ -236,6 +236,65 @@ pub struct Order {
...
+    pub take_profit: f64,
+    /// Stop loss price
+    #[serde(deserialize_with = "string_or_number")]
+    pub stop_loss: f64,
+    /// Take profit trigger price type
+    pub tp_trigger_by: TriggerPrice,
+    /// Stop loss trigger price type
+    pub sl_trigger_by: TriggerPrice,
+    /// Position idx
+    pub position_idx: i32,
```
Comment (wassup-) [inline src/order.rs:295]: Could we make this an `i64` instead?

File: src/rest/orders.rs [hunk 2]
```diff
@@ -57,13 +57,14 @@ pub struct PlaceActiveOrderData {
...
-    pub close_on_trigger: Option<bool>,
+    pub close_on_trigger: bool,
     pub order_link_id: Option<OrderLinkId>,
     pub take_profit: Option<f64>,
     pub stop_loss: Option<f64>,
     pub tp_trigger_by: Option<TriggerPrice>,
     pub sl_trigger_by: Option<TriggerPrice>,
-    pub reduce_only: Option<bool>,
+    pub reduce_only: bool,
+    pub position_idx: Option<i32>,
```
Comment (wassup-) [inline src/rest/orders.rs:67]: Could we make this an `Option<i64>` instead?

File: src/rest/orders.rs [hunk 3]
```diff
@@ -227,8 +243,9 @@ mod request {
         pub tp_trigger_by: Option<TriggerPrice>,
         #[serde(skip_serializing_if = "Option::is_none")]
         pub sl_trigger_by: Option<TriggerPrice>,
+        pub reduce_only: bool,
         #[serde(skip_serializing_if = "Option::is_none")]
-        pub reduce_only: Option<bool>,
+        pub position_idx: Option<i32>,
```
Comment (wassup-) [inline src/rest/orders.rs:248]: Could we make this an `Option<i64>` instead?

**Review by wassup- — CHANGES_REQUESTED** [review]
Thank you! I'd just like to see some really small changes and then this is ready to merge.

File: src/rest/orders.rs (hunk 2 above)
Comment (peepopoggers) [reply src/rest/orders.rs:67]: yes but the linear order requires it as mandatory. I will revert this and the other non-optional

**Comment by peepopoggers** [comment]
Fixed suggestions

**Pull request merged**